### ISSUE TITLE
Console output missing after a strange filename?

### ISSUE THREAD

**Issue by JohnVeness** [saramibreak/DiscImageCreator#129]
**Version**
20220301

**Describe the bug**
During the checking protection stage, it printed "Detected a protected file [MacroMixð" (including that odd character at the end) and then seemingly nothing else for a long time. I thought the program had hung but it turned out that it was busy creating the .scm, just not displaying anything on the console, no progress report counting through the LBAs etc. Eventually, after 30 minutes or so, some more text appeared, starting with "FILE".

**Screenshots**
```
PS C:\Users\User\Downloads> .\DiscImageCreator_20220301\Release_ANSI\DiscImageCreator.exe cd E "VC2_ANFR\VC2_ANFR.bin" 72 /c2 20 /ns /sf
AppVersion
        32 bit, AnsiBuild, 20220301T213909
/c2 val2 was omitted. set [0]
/sf val was omitted. set [60]
C:\Users\User\Downloads\VC2_ANFR\ doesn't exist, so create.
CurrentDirectory
        C:\Users\User\Downloads
WorkingPath
         Argument: VC2_ANFR\VC2_ANFR.bin
         FullPath: C:\Users\User\Downloads\VC2_ANFR\VC2_ANFR.bin
            Drive: C:
        Directory: \Users\User\Downloads\VC2_ANFR\
         Filename: VC2_ANFR
        Extension: .bin
StartTime: 2022-04-29T13:34:25+0100
Set the drive speed: 12700KB/sec
DiskSize of [C:\Users\User\Downloads\VC2_ANFR]
        Total:   100000591872 bytes
         Used:    49675403264 bytes
        ---------------------------
        Space:    50325188608 bytes
         => There is enough disk space for dumping
This drive supports [OpCode: 0xd8, SubCode: 0]
This drive supports [OpCode: 0xd8, SubCode: 1]
This drive supports [OpCode: 0xd8, SubCode: 2]
This drive supports [OpCode: 0xd8, SubCode: 8]
Checking reading lead-out -> OK
Checking SubQ adr (Track)  1/ 1
Checking SubRtoW (Track)  1/ 1
Checking Pregap sync, msf, mode (LBA)  -2788
Reading DirectoryRecord  106/ 106
Checking EXE   79
Detected Microsoft Cabinet File: \SONOW\IE\ADVAUTH.CAB
/mscf is needed to extract it
Checking EXE   80
Detected Microsoft Cabinet File: \SONOW\IE\AXA2.CAB
/mscf is needed to extract it
Checking EXE   81
Detected Microsoft Cabinet File: \SONOW\IE\BRANDING.CAB
/mscf is needed to extract it
Checking EXE   83
Detected Microsoft Cabinet File: \SONOW\IE\DXDDEX.CAB
/mscf is needed to extract it
Checking EXE   84
Detected Microsoft Cabinet File: \SONOW\IE\DXMINI.CAB
/mscf is needed to extract it
Checking EXE   86
Detected Microsoft Cabinet File: \SONOW\IE\GSETUP95.CAB
/mscf is needed to extract it
Checking EXE   87
Detected Microsoft Cabinet File: \SONOW\IE\GSETUPNT.CAB
/mscf is needed to extract it
Checking EXE   88
Detected Microsoft Cabinet File: \SONOW\IE\HHUPD.CAB
/mscf is needed to extract it
Checking EXE   89
Detected Microsoft Cabinet File: \SONOW\IE\ICW.CAB
/mscf is needed to extract it
Checking EXE   90
Detected Microsoft Cabinet File: \SONOW\IE\ICWCON.CAB
/mscf is needed to extract it
Checking EXE   91
Detected Microsoft Cabinet File: \SONOW\IE\IE4MFC40.CAB
/mscf is needed to extract it
Checking EXE   95
Detected Microsoft Cabinet File: \SONOW\IE\IECIF.CAB
/mscf is needed to extract it
Checking EXE   96
Detected Microsoft Cabinet File: \SONOW\IE\IEDATAJA.CAB
/mscf is needed to extract it
Checking EXE   97
Detected Microsoft Cabinet File: \SONOW\IE\IENT_S1.CAB
/mscf is needed to extract it
Checking EXE   98
Detected Microsoft Cabinet File: \SONOW\IE\IENT_S2.CAB
/mscf is needed to extract it
Checking EXE   99
Detected Microsoft Cabinet File: \SONOW\IE\IENT_S3.CAB
/mscf is needed to extract it
Checking EXE  100
Detected Microsoft Cabinet File: \SONOW\IE\IENT_S4.CAB
/mscf is needed to extract it
Checking EXE  101
Detected Microsoft Cabinet File: \SONOW\IE\IENT_S5.CAB
/mscf is needed to extract it
Checking EXE  102
Detected Microsoft Cabinet File: \SONOW\IE\IEW2K_1.CAB
/mscf is needed to extract it
Checking EXE  103
Detected Microsoft Cabinet File: \SONOW\IE\IEW2K_2.CAB
/mscf is needed to extract it
Checking EXE  104
Detected Microsoft Cabinet File: \SONOW\IE\IEW2K_3.CAB
/mscf is needed to extract it
Checking EXE  105
Detected Microsoft Cabinet File: \SONOW\IE\IE_S1.CAB
/mscf is needed to extract it
Checking EXE  106
Detected Microsoft Cabinet File: \SONOW\IE\IE_S2.CAB
/mscf is needed to extract it
Checking EXE  107
Detected Microsoft Cabinet File: \SONOW\IE\IE_S3.CAB
/mscf is needed to extract it
Checking EXE  108
Detected Microsoft Cabinet File: \SONOW\IE\IE_S4.CAB
/mscf is needed to extract it
Checking EXE  109
Detected Microsoft Cabinet File: \SONOW\IE\MOBILE95.CAB
/mscf is needed to extract it
Checking EXE  110
Detected Microsoft Cabinet File: \SONOW\IE\MOBILENT.CAB
/mscf is needed to extract it
Checking EXE  111
Detected Microsoft Cabinet File: \SONOW\IE\MSN_AUTH.CAB
/mscf is needed to extract it
Checking EXE  113
Detected Microsoft Cabinet File: \SONOW\IE\SETUPNT.CAB
/mscf is needed to extract it
Checking EXE  114
Detected Microsoft Cabinet File: \SONOW\IE\SETUPW95.CAB
/mscf is needed to extract it
Checking EXE  115
Detected Microsoft Cabinet File: \SONOW\IE\TS95.CAB
/mscf is needed to extract it
Checking EXE  116
Detected Microsoft Cabinet File: \SONOW\IE\TSNT.CAB
/mscf is needed to extract it
Checking EXE  117
Detected Microsoft Cabinet File: \SONOW\IE\VMX86_01.CAB
/mscf is needed to extract it
Checking EXE  118
Detected Microsoft Cabinet File: \SONOW\IE\VMX86_02.CAB
/mscf is needed to extract it
Checking EXE  119 SWSETUP.EXE: File Version 8.0

Detected Wise Installation: \SONOW\SW\SWSETUP.EXE
Please wait until all files are extracted. This is needed to search protection
Checking EXE  213
Detected a protected file [MacroMixðFILE: C:\Users\User\Downloads\VC2_ANFR\VC2_ANFR.img
Checking sectors: 324081/324081
[ERROR] Number of sector(s) where user data doesn't match the expected ECC/EDC: 24
[ERROR] Number of sector(s) where sync(0x00 - 0x0c) is invalid: 2
Total errors: 26
Total warnings: 0
Creating cue and ccd (Track)  1/ 1
Creating bin (Track)  1/ 1
Hashing: VC2_ANFR.scm
Hashing: VC2_ANFR.img
Hashing: VC2_ANFR.bin
EndTime: 2022-04-29T14:17:28+0100
```

**Disc title**
Virtual Chess 2 (Sold Out Software)
![DiscFront](https://user-images.githubusercontent.com/2512915/165952912-bb1a877d-854f-4011-80c5-3235bf5123cf.jpg)

![DiscBack](https://user-images.githubusercontent.com/2512915/165953156-2ebd1500-f496-4495-9b84-6e186c13d2be.jpg)

(I only have the disc).

**Disc ringcode**
B20377 SOC000235/THE PRODUCERS 21/09/2004

IFPI LP57

**URL**
https://www.ebay.co.uk/itm/144073951254

Or I can send you my disc - I don't need it.

**Log file**
https://mega.nz/file/wxsE0BQQ#MOo2nFy06_5oH6pE7Pu-cBA96NzFzOu8gs6mCNRL4uk

**Comment by saramibreak** [comment]
> "Detected a protected file [MacroMixð" (including that odd character at the end)

Uploaded test version.
[DiscImageCreator_test.zip](https://github.com/saramibreak/DiscImageCreator/files/8598624/DiscImageCreator_test.zip)

Try, please.

**Comment by JohnVeness** [comment]
Thanks, that seems to have fixed it.

Output from same disc:
```
PS C:\Users\User\Downloads> .\DiscImageCreator_test\Release_ANSI\DiscImageCreator.exe cd E "VC2_ANFR\VC2_ANFR.bin" 72 /c2 20 /ns /sf
AppVersion
        32 bit, AnsiBuild, 20220501T182336
/c2 val2 was omitted. set [0]
/sf val was omitted. set [60]
C:\Users\User\Downloads\VC2_ANFR\ doesn't exist, so create.
CurrentDirectory
        C:\Users\User\Downloads
WorkingPath
         Argument: VC2_ANFR\VC2_ANFR.bin
         FullPath: C:\Users\User\Downloads\VC2_ANFR\VC2_ANFR.bin
            Drive: C:
        Directory: \Users\User\Downloads\VC2_ANFR\
         Filename: VC2_ANFR
        Extension: .bin
StartTime: 2022-05-03T21:05:53+0100
Set the drive speed: 12700KB/sec
DiskSize of [C:\Users\User\Downloads\VC2_ANFR]
        Total:   100000591872 bytes
         Used:    47449821184 bytes
        ---------------------------
        Space:    52550770688 bytes
         => There is enough disk space for dumping
This drive supports [OpCode: 0xd8, SubCode: 0]
This drive supports [OpCode: 0xd8, SubCode: 1]
This drive supports [OpCode: 0xd8, SubCode: 2]
This drive supports [OpCode: 0xd8, SubCode: 8]
Checking reading lead-out -> OK
Checking SubQ adr (Track)  1/ 1
Checking SubRtoW (Track)  1/ 1
Checking Pregap sync, msf, mode (LBA)  -3083
Reading DirectoryRecord  106/ 106
Checking EXE   79
Detected Microsoft Cabinet File: \SONOW\IE\ADVAUTH.CAB
/mscf is needed to extract it
Checking EXE   80
Detected Microsoft Cabinet File: \SONOW\IE\AXA2.CAB
/mscf is needed to extract it
Checking EXE   81
Detected Microsoft Cabinet File: \SONOW\IE\BRANDING.CAB
/mscf is needed to extract it
Checking EXE   83
Detected Microsoft Cabinet File: \SONOW\IE\DXDDEX.CAB
/mscf is needed to extract it
Checking EXE   84
Detected Microsoft Cabinet File: \SONOW\IE\DXMINI.CAB
/mscf is needed to extract it
Checking EXE   86
Detected Microsoft Cabinet File: \SONOW\IE\GSETUP95.CAB
/mscf is needed to extract it
Checking EXE   87
Detected Microsoft Cabinet File: \SONOW\IE\GSETUPNT.CAB
/mscf is needed to extract it
Checking EXE   88
Detected Microsoft Cabinet File: \SONOW\IE\HHUPD.CAB
/mscf is needed to extract it
Checking EXE   89
Detected Microsoft Cabinet File: \SONOW\IE\ICW.CAB
/mscf is needed to extract it
Checking EXE   90
Detected Microsoft Cabinet File: \SONOW\IE\ICWCON.CAB
/mscf is needed to extract it
Checking EXE   91
Detected Microsoft Cabinet File: \SONOW\IE\IE4MFC40.CAB
/mscf is needed to extract it
Checking EXE   95
Detected Microsoft Cabinet File: \SONOW\IE\IECIF.CAB
/mscf is needed to extract it
Checking EXE   96
Detected Microsoft Cabinet File: \SONOW\IE\IEDATAJA.CAB
/mscf is needed to extract it
Checking EXE   97
Detected Microsoft Cabinet File: \SONOW\IE\IENT_S1.CAB
/mscf is needed to extract it
Checking EXE   98
Detected Microsoft Cabinet File: \SONOW\IE\IENT_S2.CAB
/mscf is needed to extract it
Checking EXE   99
Detected Microsoft Cabinet File: \SONOW\IE\IENT_S3.CAB
/mscf is needed to extract it
Checking EXE  100
Detected Microsoft Cabinet File: \SONOW\IE\IENT_S4.CAB
/mscf is needed to extract it
Checking EXE  101
Detected Microsoft Cabinet File: \SONOW\IE\IENT_S5.CAB
/mscf is needed to extract it
Checking EXE  102
Detected Microsoft Cabinet File: \SONOW\IE\IEW2K_1.CAB
/mscf is needed to extract it
Checking EXE  103
Detected Microsoft Cabinet File: \SONOW\IE\IEW2K_2.CAB
/mscf is needed to extract it
Checking EXE  104
Detected Microsoft Cabinet File: \SONOW\IE\IEW2K_3.CAB
/mscf is needed to extract it
Checking EXE  105
Detected Microsoft Cabinet File: \SONOW\IE\IE_S1.CAB
/mscf is needed to extract it
Checking EXE  106
Detected Microsoft Cabinet File: \SONOW\IE\IE_S2.CAB
/mscf is needed to extract it
Checking EXE  107
Detected Microsoft Cabinet File: \SONOW\IE\IE_S3.CAB
/mscf is needed to extract it
Checking EXE  108
Detected Microsoft Cabinet File: \SONOW\IE\IE_S4.CAB
/mscf is needed to extract it
Checking EXE  109
Detected Microsoft Cabinet File: \SONOW\IE\MOBILE95.CAB
/mscf is needed to extract it
Checking EXE  110
Detected Microsoft Cabinet File: \SONOW\IE\MOBILENT.CAB
/mscf is needed to extract it
Checking EXE  111
Detected Microsoft Cabinet File: \SONOW\IE\MSN_AUTH.CAB
/mscf is needed to extract it
Checking EXE  113
Detected Microsoft Cabinet File: \SONOW\IE\SETUPNT.CAB
/mscf is needed to extract it
Checking EXE  114
Detected Microsoft Cabinet File: \SONOW\IE\SETUPW95.CAB
/mscf is needed to extract it
Checking EXE  115
Detected Microsoft Cabinet File: \SONOW\IE\TS95.CAB
/mscf is needed to extract it
Checking EXE  116
Detected Microsoft Cabinet File: \SONOW\IE\TSNT.CAB
/mscf is needed to extract it
Checking EXE  117
Detected Microsoft Cabinet File: \SONOW\IE\VMX86_01.CAB
/mscf is needed to extract it
Checking EXE  118
Detected Microsoft Cabinet File: \SONOW\IE\VMX86_02.CAB
/mscf is needed to extract it
Checking EXE  119 SWSETUP.EXE: File Version 8.0

Detected Wise Installation: \SONOW\SW\SWSETUP.EXE
Please wait until all files are extracted. This is needed to search protection
Checking EXE  213
Detected a protected file [MacroMix]. LBA 0 to 0
Set OpCode: 0xd8, SubCode: 8(Raw)
[INFO] SecuROM sector not found
Checking SubQ ctl (Track)  1/ 1
Creating .scm (LBA) 217201/324082 LBA[217202, 0x35072] Detected C2 error 25 bit
Creating .scm (LBA) 217202/324082 LBA[217203, 0x35073] Detected C2 error 21 bit
Creating .scm (LBA) 217219/324082 LBA[217220, 0x35084] Detected C2 error 46 bit
Creating .scm (LBA) 217220/324082 LBA[217221, 0x35085] Detected C2 error 34 bit
Creating .scm (LBA) 217237/324082 LBA[217238, 0x35096] Detected C2 error 49 bit
Creating .scm (LBA) 217238/324082 LBA[217239, 0x35097] Detected C2 error 31 bit
Creating .scm (LBA) 217255/324082 LBA[217256, 0x350a8] Detected C2 error 45 bit
Creating .scm (LBA) 217256/324082 LBA[217257, 0x350a9] Detected C2 error 16 bit
Creating .scm (LBA) 217273/324082 LBA[217274, 0x350ba] Detected C2 error 41 bit
Creating .scm (LBA) 217274/324082 LBA[217275, 0x350bb] Detected C2 error 21 bit
Creating .scm (LBA) 217291/324082 LBA[217292, 0x350cc] Detected C2 error 50 bit
Creating .scm (LBA) 217292/324082 LBA[217293, 0x350cd] Detected C2 error 16 bit
Creating .scm (LBA) 217309/324082 LBA[217310, 0x350de] Detected C2 error 47 bit
Creating .scm (LBA) 217310/324082 LBA[217311, 0x350df] Detected C2 error 18 bit
Creating .scm (LBA) 217327/324082 LBA[217328, 0x350f0] Detected C2 error 71 bit
Creating .scm (LBA) 217328/324082 LBA[217329, 0x350f1] Detected C2 error 10 bit
Creating .scm (LBA) 217345/324082 LBA[217346, 0x35102] Detected C2 error 90 bit
Creating .scm (LBA) 217346/324082 LBA[217347, 0x35103] Detected C2 error 13 bit
Creating .scm (LBA) 217363/324082 LBA[217364, 0x35114] Detected C2 error 50 bit
Creating .scm (LBA) 217364/324082 LBA[217365, 0x35115] Detected C2 error 6 bit
Creating .scm (LBA) 217381/324082 LBA[217382, 0x35126] Detected C2 error 86 bit
Creating .scm (LBA) 217382/324082 LBA[217383, 0x35127] Detected C2 error 9 bit
Creating .scm (LBA) 217399/324082 LBA[217400, 0x35138] Detected C2 error 99 bit
Creating .scm (LBA) 217400/324082 LBA[217401, 0x35139] Detected C2 error 6 bit
Creating .scm (LBA) 217417/324082 LBA[217418, 0x3514a] Detected C2 error 101 bit
Creating .scm (LBA) 217418/324082 LBA[217419, 0x3514b] Detected C2 error 7 bit
Creating .scm (LBA) 217435/324082 LBA[217436, 0x3515c] Detected C2 error 39 bit
Creating .scm (LBA) 217436/324082 LBA[217437, 0x3515d] Detected C2 error 2 bit
Creating .scm (LBA) 217453/324082 LBA[217454, 0x3516e] Detected C2 error 102 bit
Creating .scm (LBA) 217454/324082 LBA[217455, 0x3516f] Detected C2 error 2 bit
Creating .scm (LBA) 217471/324082 LBA[217472, 0x35180] Detected C2 error 56 bit
Creating .scm (LBA) 217489/324082 LBA[217490, 0x35192] Detected C2 error 63 bit
Creating .scm (LBA) 217507/324082 LBA[217508, 0x351a4] Detected C2 error 80 bit
Creating .scm (LBA) 217525/324082 LBA[217526, 0x351b6] Detected C2 error 98 bit
Creating .scm (LBA) 217543/324082 LBA[217544, 0x351c8] Detected C2 error 80 bit
Creating .scm (LBA) 217561/324082 LBA[217562, 0x351da] Detected C2 error 94 bit
Creating .scm (LBA) 217579/324082 LBA[217580, 0x351ec] Detected C2 error 89 bit
Creating .scm (LBA) 217597/324082 LBA[217598, 0x351fe] Detected C2 error 80 bit
Creating .scm (LBA) 217615/324082 LBA[217616, 0x35210] Detected C2 error 94 bit
Creating .scm (LBA) 217633/324082 LBA[217634, 0x35222] Detected C2 error 89 bit
Creating .scm (LBA) 217651/324082 LBA[217652, 0x35234] Detected C2 error 104 bit
Creating .scm (LBA) 217903/324082 LBA[217904, 0x35330] Detected C2 error 24 bit
Creating .scm (LBA) 217904/324082 LBA[217905, 0x35331] Detected C2 error 16 bit
Creating .scm (LBA) 217921/324082 LBA[217922, 0x35342] Detected C2 error 32 bit
Creating .scm (LBA) 217922/324082 LBA[217923, 0x35343] Detected C2 error 23 bit
Creating .scm (LBA) 218029/324082 LBA[218030, 0x353ae] Detected C2 error 20 bit
Creating .scm (LBA) 218030/324082 LBA[218031, 0x353af] Detected C2 error 19 bit
Creating .scm (LBA) 218047/324082 LBA[218048, 0x353c0] Detected C2 error 49 bit
Creating .scm (LBA) 218048/324082 LBA[218049, 0x353c1] Detected C2 error 51 bit
Creating .scm (LBA) 218065/324082 LBA[218066, 0x353d2] Detected C2 error 49 bit
Creating .scm (LBA) 218066/324082 LBA[218067, 0x353d3] Detected C2 error 26 bit
Creating .scm (LBA) 218119/324082 LBA[218120, 0x35408] Detected C2 error 49 bit
Creating .scm (LBA) 218120/324082 LBA[218121, 0x35409] Detected C2 error 51 bit
Creating .scm (LBA) 310982/324082 LBA[310983, 0x4bec7] Detected C2 error 11 bit
Creating .scm (LBA) 311333/324082 LBA[311334, 0x4c026] Detected C2 error 11 bit
Creating .scm (LBA) 312181/324082 LBA[312182, 0x4c376] Detected C2 error 22 bit
Creating .scm (LBA) 318272/324082 LBA[318273, 0x4db41] Detected C2 error 21 bit
Creating .scm (LBA) 318648/324082 LBA[318649, 0x4dcb9] Detected C2 error 13 bit
Creating .scm (LBA) 319651/324082 LBA[319652, 0x4e0a4] Detected C2 error 18 bit
Creating .scm (LBA) 324082/324082
Need to reread sector: 319652 rereading times:    1/  20
Done. See _c2Error.txt
Copying .scm to .img
Descrambling data sector of img: 324081/324081
Exec ""C:\Users\User\Downloads\DiscImageCreator_test\Release_ANSI\EccEdc.exe" check "C:\Users\User\Downloads\VC2_ANFR\VC2_ANFR.img""
FILE: C:\Users\User\Downloads\VC2_ANFR\VC2_ANFR.img
Checking sectors: 324081/324081
[ERROR] Number of sector(s) where user data doesn't match the expected ECC/EDC: 24
[ERROR] Number of sector(s) where sync(0x00 - 0x0c) is invalid: 2
Total errors: 26
Total warnings: 0
Creating cue and ccd (Track)  1/ 1
Creating bin (Track)  1/ 1
Hashing: VC2_ANFR.scm
Hashing: VC2_ANFR.img
Hashing: VC2_ANFR.bin
EndTime: 2022-05-03T21:50:09+0100
```

(Obviously the disc itself isn't OK, but the program output now is!)

Many thanks again.

**Comment by saramibreak** [comment]
978aa8067cdda7b09ea21224331fb4846bfe111f

**Comment by JohnVeness** [comment]
Hi. This issue still seems to be present in 20220606. Was it not merged to that release?

**Comment by JohnVeness** [comment]
For the record, this is now fixed in 20220707, thanks :)
